### PR TITLE
Add directories example

### PR DESCRIPTION
**Pull request by eliben**
Note: there are other topics we may want to cover that may or may not belong in the same example:

1. `os.File` objects
2. Temporary files and directories
3. `filepath.Walk` for recursive traversal

I'm not sure how to incorporate (1). For (2), I think it makes sense to create a separate example because it covers both files and directories and has its own topic. For (3), it may actually make sense to just add a snippet to *this* example, because we're already building a dir hierarchy and listing contents non-recursively with `ReadDir`, so `Walk` seems to belong naturally.

WDYT?

### CODE REVIEW

File: examples/directories/directories.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	// Helper function to create a new empty file.
+	createEmptyFile := func(name string) {
+		d := []byte("")
+		check(ioutil.WriteFile(name, d, 0644))
+	}
+
+	createEmptyFile("subdir/file1")
+
+	// We can create a hierarchy of directories, including
+	// parents wiht `MkdirAll`. This is similar to the
```
Comment (mmcgrana) [inline examples/directories/directories.go]:
```suggestion
	// parents with `MkdirAll`. This is similar to the
```

**Comment by mmcgrana** [comment]
> 2. Temporary files and directories
> 3. `filepath.Walk` for recursive traversal
> 
> I'm not sure how to incorporate (1). For (2), I think it makes sense to create a separate example because it covers both files and directories and has its own topic. For (3), it may actually make sense to just add a snippet to _this_ example, because we're already building a dir hierarchy and listing contents non-recursively with `ReadDir`, so `Walk` seems to belong naturally.

Agree on both (2) and (3).

For (1), my intuition is just to cover it where needed in the context of more usage-oriented examples.

**Comment by mmcgrana** [comment]
This example looks good mod the one typo - thanks!

**Comment by eliben** [comment]
Done w.r.t. (3) - added it here, PTAL. Note about the implementation: I made `visit` a separate function mainly because of the width limitation, squeezing it on the same line with the call to `Walk` is challenging.

For (2) I'll create an issue to track.

**Comment by mmcgrana** [comment]
Great! I've merged this one and the temporary files and directory example too, with minor cosmetic tweaks. Thanks for making these and sorry it took a while to get back to you on them!

**Pull request merged**